doc/design/python_interface.md

```diff
@@ -0,0 +1,101 @@
+# Python Interface Design
+
+## Variable as variable data type (not parameter)
+The class Variable is introduced to define the variables that are not parameters.
+
+All the variables are defined in a variable scope, `VarScope`, it will check duplicate variable names.
```

Comment:
It seems that `Scope` is enough?

Reply:
this design is based on current implementation ?

Reply:
Not really. This design should handle current implementation and future compile time concepts.

However, is the compile time concept of `Scope` needs a different name? because in that time, the user cannot access runtime scope directly, so there is only one concept which related to scope in future?